nats.go

```diff
@@ -2507,6 +2507,19 @@ func (nc *Conn) IsConnected() bool {
 	return nc.isConnected()
 }
 
+// Servers returns the list of known servers, including additional
```

Comment:
Maybe we should tag servers in a structure that denotes whether they were explicit or implicit? We should be sensitive to credentials IMO. 

How do we handle applying credentials to the implicit, sever delivered, information?